### DISCUSSION TITLE
[Feature Request] Transformations: Join By Field (Inner Join)

### DISCUSSION THREAD

**Discussion by AlexKaracaoglu** [grafana/grafana#53835] [Dashboards, Panels & Visualizations]
# New Feature Request - Inner Join Transformation
### Idea
Currently, when it comes to joining across different data sources using transformations, there is only an outer join or a merge option. Starting a discussion to propose adding an inner join transformation. There are some workarounds using multiple transformations to simulate this (ie. and outer join or merge + a follow up filter transformation), but a one stop shop to perform an inner join would be great.

### Use Case
Use this transformation to combine the result from multiple queries (combining on a passed join field or the first time column) into one single result and drop rows where a successful join isn't able to occur - performing an inner join.

### Example
In the example below, we have two queries returning table data. It is visualized as two separate tables before applying the inner join transformation.

Query A:
Time | Job | Uptime
-- | -- | --
2020-07-07 11:34:20 | node | 25260122
2020-07-07 11:24:20 | postgre | 123001233
2020-07-07 11:14:20 | postgre | 345001233

Query B:
Time | Server | Errors
-- | -- | --
2020-07-07 11:34:20 | server 1 | 15
2020-07-07 11:24:20 | server 2 | 5
2020-07-07 11:04:20 | server 3 | 10

Result after applying the inner join transformation:
Time | Job | Uptime | Server | Errors
-- | -- | -- | -- | --
2020-07-07 11:34:20 | node | 25260122 | server 1 | 15
2020-07-07 11:24:20 | postgre | 123001233 | server 2 | 5

### Next Steps
Happy to discuss! And if there is interest in this feature, I have code ready to add it so I'd be happy to open a PR.

**Comment by AlexKaracaoglu** [comment]
Ah, didn't see it at first, looks like a duplicate of https://github.com/grafana/grafana/discussions/38150

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
